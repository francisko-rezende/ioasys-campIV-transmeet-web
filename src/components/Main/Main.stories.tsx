import React from 'react'

import { ComponentMeta, ComponentStory } from '@storybook/react'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
} as ComponentMeta<typeof Main>

const Template: ComponentStory<typeof Main> = (args) => <Main {...args} />

export const Basic = Template.bind({})
