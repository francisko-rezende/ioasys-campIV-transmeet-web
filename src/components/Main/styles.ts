import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  padding: 3rem;
`

export const Title = styled.h1`
  padding: 1rem;
`

export const Description = styled.h2``

export const RequisitionsWrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
`
