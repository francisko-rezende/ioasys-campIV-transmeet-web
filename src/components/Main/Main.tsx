import React from 'react'

import * as S from './styles'

const Main = ({
  title = 'CRA Project Starter',
  description = 'A CRA template that includes ESLINT/Prettier, Styled Components and Storybook',
}) => {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
    </S.Wrapper>
  )
}

export default Main
