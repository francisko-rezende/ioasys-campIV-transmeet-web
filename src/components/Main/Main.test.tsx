import React from 'react'

import { render, screen } from '@testing-library/react'

import Main from '.'

test('that Main renders properly', () => {
  render(<Main />)

  expect(
    screen.getByRole('heading', { name: /cra project starter/i }),
  ).toBeInTheDocument()

  expect(
    screen.getByRole('heading', {
      name: /A CRA template that includes ESLINT\/Prettier, Styled Components and Storybook/i,
    }),
  ).toBeInTheDocument()

  expect(screen.getByRole('main')).toMatchSnapshot()
})

test('that Main renders colors properly', () => {
  render(<Main />)

  expect(screen.getByRole('main')).toHaveStyle({
    'background-color': '#06092b',
    color: '#fff',
  })
})
