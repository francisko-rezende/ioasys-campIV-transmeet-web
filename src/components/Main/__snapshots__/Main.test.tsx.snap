// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`that Main renders properly 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3rem;
}

.c1 {
  padding: 1rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    CRA Project Starter
  </h1>
  <h2
    class=""
  >
    A CRA template that includes ESLINT/Prettier, Styled Components and Storybook
  </h2>
</main>
`;
